pkg/sql/schemachanger/testdata/end_to_end/alter_table_add_check_not_valid

```diff
@@ -0,0 +1,80 @@
+setup
+CREATE TABLE t (i INT PRIMARY KEY);
+INSERT INTO t VALUES (0);
+----
+...
++object {100 101 t} -> 104
+
+
+test
+ALTER TABLE t ADD CHECK (i > 0) NOT VALID;
```

Comment:
These end-to-end tests are quite expensive because they balloon into TestRollback and TestBackup tests. I don't think there's much value in having one for each three non-index constraint types, just pick one.

Unrelated but this test case is nice in that it sets the stage for a VALIDATE CONSTRAINT case to be appended to the file later on.